evap/student/views.py

```diff
@@ -60,11 +60,20 @@ def get_valid_form_groups_or_render_vote_page(request, course, preview, for_rend
 
     course_form_group = form_groups.pop(course.general_contribution)
 
+    contributor_form_groups = [(contribution.contributor, contribution.label, form_group, any(form.errors for form in form_group)) for contribution, form_group in form_groups.items()]
+    if contributor_form_groups:
+        course_form_group_top = [questions_form for questions_form in course_form_group if questions_form.questionnaire.is_above_contributors]
+        course_form_group_bottom = [questions_form for questions_form in course_form_group if questions_form.questionnaire.is_below_contributors]
+    else:
+        course_form_group_top = [questions_form for questions_form in course_form_group if questions_form.questionnaire.is_above_contributors or questions_form.questionnaire.is_below_contributors]
+        course_form_group_bottom = []
```

Comment:
i would prefer something like 
```
course_form_group_top = ...
course_form_group_bottom = ...
if not contributor_form_groups:
    course_form_group_top += course_form_group_bottom 
    course_form_group_bottom  = []
```
it's less code to parse, and is a bit more "here's the common case and here if need be some special handling on top" instead of two equal cases.